### PR TITLE
[NOJIRA] Improve rebase instructions

### PR DESCRIPTION
**Pull request by matthewdavidson**

### CODE REVIEW

File: packages/react-scripts/README.md [hunk 1]
```diff
@@ -97,16 +97,41 @@ If you decide to opt out, Sass files will not be treated as CSS Modules by defau
...
-git fetch upstream
-git rebase <commit>
-```
+1. Ensure `master` is in sync with `upstream/master`:
+
+    ```sh
+    git checkout master
+    git remote add upstream git@github.com:facebook/create-react-app.git
+    git fetch upstream
+    git pull upstream master
```
Comment (k0nserv) [inline packages/react-scripts/README.md]: We just want `master` to track `upstream/master` right? If so `git fetch upstream` followed by `git reset --hard upstream/master` is better than pull

File: packages/react-scripts/README.md [hunk 2]
```diff
@@ -97,16 +97,41 @@ If you decide to opt out, Sass files will not be treated as CSS Modules by defau
...
+
+    ```sh
+    git branch -m <branch>
+    git push origin <branch>
+    ```
+
+1. Finally, when we are confident that the rebase has been successful, re-name your branch back to `fork` and push it to origin:
+
+    ```sh
+    git branch -m fork
```
Comment (k0nserv) [inline packages/react-scripts/README.md]: If we are keeping branches like this could we not keep one per release too?

Comment (matthewdavidson) [reply packages/react-scripts/README.md]: We could ... we haven't actually done it yet so im thinking we can amend as we go.

Bear in mind, tags will still persist ...

Comment (k0nserv) [reply packages/react-scripts/README.md]: Okay let's bear that in mind. Would you say the chance of accidentally nuking our changes to the fork are increased, decreased, or the same with this approach?

Comment (matthewdavidson) [reply packages/react-scripts/README.md]: Decreased. We'll always have a reference to commits (via our tags) branching off of each of their tagged releases.

**Pull request merged**